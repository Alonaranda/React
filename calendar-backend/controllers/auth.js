const { response } = require('express');

const crearUsuario = (req, res = response) => {
    //console.log(req.body);
    const { name, email, password } = req.body;
    

    res.status(201).json({
        ok: true,
        msg: 'registro',
        name,
        email,
        password
    });
}


const loginUsuario = (req, res = response) => {

    const { email, password } = req.body;

    res.status(200).json({
        ok: true,
        msg: 'login',
        email,
        password
    
    });
}


const revalidarToken = (req, res = response) => {
    res.json({
        ok: true,
        msg: 'renew'
    });
}


//Exportaciones
module.exports = {
    crearUsuario,
    loginUsuario,
    revalidarToken
}