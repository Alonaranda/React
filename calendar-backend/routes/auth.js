/*
    Rutas de Usuarios || Auth
    host + /api/auth + ...
 */
//Status --> https://www.restapitutorial.com/httpstatuscodes.html

const { Router } = require('express');
const { check} = require('express-validator');
const { validarCampos } = require('../middlewares/validar-campos');
const { crearUsuario, loginUsuario, revalidarToken } = require('../controllers/auth');
const { validarJWT } = require('../middlewares/validar-jwt');

const router = Router();



//Ruta de registro - POST
router.post(
    '/new', 
    [// Middlewares
        check('name', 'Nombre es obligatorio').not().isEmpty(),
        check('email', 'Correo es obligatorio').isEmail(),
        check('password', 'Password es obligatorio, mayor a 6 char').isLength({min: 6}),
        validarCampos
    ], 
    crearUsuario
);

//Ruta de login - POST
router.post(
    '/', 
    [
        check('email', 'Correo es obligatorio').isEmail(),
        check('password', 'Password es obligatorio, mayor a 6 char').isLength({min: 6}),
        validarCampos
    ], 
    loginUsuario
);

//Ruta de validacion de token - GET
router.get('/renew', validarJWT , revalidarToken);


//Exportaciones
module.exports = router;