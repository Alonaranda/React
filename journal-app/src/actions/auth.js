//Simples funciones
import { types } from "../types/types";
import {firebase, googleAuthProvider} from '../firebase/firebase-config'
import { startLoading, finishLoading } from "./ui";

//Funcion asincrona, cualquier funcion 
export const startLoginEmailPassword = (email, password) => {
    return (dispatch) => {
        // setTimeout(() => {
        //     dispatch(login;(123, 'Pedro'));
        // }, 3000);
        dispatch(startLoading())
        firebase.auth().signInWithEmailAndPassword(email, password)
        .then(({user}) => {
            dispatch(
                login(user.uid, user.displayName)
            )
            dispatch(finishLoading());
        })
        .catch(({message}) => {
            console.log(message);
            dispatch(finishLoading())
        })
    }
}

//Si es funcion asincrona necesito regresar un callback
export const startRegisterWithEmailPasswordName = (email, password, name) => {
    return (dispatch) => {
        firebase.auth().createUserWithEmailAndPassword(email, password)
        .then(async({user}) => {
            await user.updateProfile({displayName: name});
            //console.log(user);
            dispatch(
                login(user.uid, user.displayName)
            )
        })
        .catch(({message}) => {
            console.log(message);
        })
        console.log('Registrado!');
    }
}

//Action de login con google
export const startGoogleLogin = () => {
    return (dispatch) => {
        firebase.auth().signInWithPopup(googleAuthProvider)
            .then(({user}) => {
                dispatch(
                    login(user.uid, user.displayName)
                )
            })
    }
}


//Manda lo que nos interesa el id y nombre
export const login = (uid, displayName) => {
    return {
        type: types.login,
        payload: {
            uid,
            displayName
        }
    }
}
