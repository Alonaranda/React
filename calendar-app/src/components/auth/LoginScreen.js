import React from 'react';
import {useDispatch} from 'react-redux'
import './login.css';
import { useForm } from '../../hooks/useForm';
import { startLogin, startRegister } from '../../actions/auth';
import Swal from 'sweetalert2';

export const LoginScreen = () => {

    const dispatch = useDispatch();


    //Form Login
    const [formLoginValues, handleInputChangeLogin] = useForm({
        lEmail: 'chris@gmail.com',
        lPassword: '123456'
    });
    const {lEmail, lPassword} = formLoginValues;
    const handleLogin = (e) => {
        e.preventDefault();
        //console.log(formLoginValues);
        dispatch(startLogin(lEmail,lPassword));
    }


    //Form Register
    const [formRegisterValues, handleInputChangeRegister] = useForm({
        rName: 'Alonso',
        rEmail: 'alonso@gmail.com',
        rPassword1: '123456',
        rPassword2: '123456',
    });
    const {rName, rEmail, rPassword1, rPassword2} = formRegisterValues;
    const hanldeRegister = (e) => {
        e.preventDefault();
        console.log(formRegisterValues);
        if(rPassword1 !== rPassword2){
            return Swal.fire('Error', 'Las contrase単as no coinciden', 'error');
        }
        dispatch(startRegister(rEmail, rPassword1, rName));
    } 



    return (
        <div className="container login-container">
            <div className="row">
                <div className="col-md-6 login-form-1">
                    <h3>Ingreso</h3>
                    <form onSubmit={handleLogin}>
                        <div className="form-group">
                            <input 
                                type="text"
                                className="form-control"
                                placeholder="Correo"
                                name="lEmail"
                                value={lEmail}
                                onChange={handleInputChangeLogin}
                            />
                        </div>
                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Contrase単a"
                                name="lPassword"
                                value={lPassword}
                                onChange={handleInputChangeLogin}
                            />
                        </div>
                        <div className="form-group">
                            <input 
                                type="submit"
                                className="btnSubmit"
                                value="Login" 
                            />
                        </div>
                    </form>
                </div>

                <div className="col-md-6 login-form-2">
                    <h3>Registro</h3>
                    <form onSubmit={hanldeRegister}>
                        <div className="form-group">
                            <input
                                type="text"
                                className="form-control"
                                placeholder="Nombre"
                                name="rName"
                                value={rName}
                                onChange={handleInputChangeRegister}
                            />
                        </div>
                        <div className="form-group">
                            <input
                                type="email"
                                className="form-control"
                                placeholder="Correo"
                                name="rEmail"
                                value={rEmail}
                                onChange={handleInputChangeRegister}
                            />
                        </div>
                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Contrase単a" 
                                name="rPassword1"
                                value={rPassword1}
                                onChange={handleInputChangeRegister}
                            />
                        </div>

                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Repita la contrase単a" 
                                name="rPassword2"
                                value={rPassword2}
                                onChange={handleInputChangeRegister}
                            />
                        </div>

                        <div className="form-group">
                            <input 
                                type="submit" 
                                className="btnSubmit" 
                                value="Crear cuenta" />
                        </div>
                    </form>
                </div>
            </div>
        </div>
    )
}