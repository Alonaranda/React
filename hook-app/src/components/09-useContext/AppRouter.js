import React from 'react';
//Importaciones para usar el Router
import { NavBar } from './NavBar';
import {
    BrowserRouter as Router,
    Switch,
    Route,
  } from "react-router-dom";
import {AboutScreen} from './AboutScreen';
import {LoginScreen} from './LoginScreen';
import {HomeScreen} from './HomeScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div className="container">
                <NavBar/>

                {/* Switch de los Routers*/}
                <Switch>
                    <Route exact path="/about" component={AboutScreen}/>
                    <Route exact path="/login" component={LoginScreen}/>
                    <Route exact path="/" component={HomeScreen}/>
                    <Route path="*" component={HomeScreen}/>
                    {/*Se puede usar el Redirect */}
                </Switch>
            </div>
        </Router>
    )
}
