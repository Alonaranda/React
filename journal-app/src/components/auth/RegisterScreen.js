import React from "react";
import { Link } from "react-router-dom";
import { useForm } from "../../hooks/useForms";
//import validator from 'validator';
import {useDispatch, useSelector} from 'react-redux';
import { setError, removeError } from "../../actions/ui";
import { startRegisterWithEmailPasswordName } from "../../actions/auth";


export const RegisterScreen = () => {

  const dispatch = useDispatch();
  const {msgError} = useSelector(state => state.ui);

  //Obtener valores de mi form
  const [formValues, handleInputChange] = useForm({
    name: 'Christian',
    email: 'admin@admin.com',
    password: '123456',
    password2: '123456'
  });
  const {name, email, password, password2} = formValues;

  //Submit
  const handleRegister = (e) => {
    e.preventDefault();
    if(isFormValid()){
      //console.log('Form correcto');
      dispatch(startRegisterWithEmailPasswordName(email, password, name));
    }

    
  }

  const isFormValid = () => {
      if(name.trim().length === 0){
        //console.log('Name is required');
        dispatch(setError('Name is required'));
        return false;
      }else if(email.trim().length === 0){
        //console.log('Email not valid');
        dispatch(setError('Email not valid'));
        return false;
      }else if(password !== password2 || password.length < 5){
        //console.log('Mejora tu password');
        dispatch(setError('Revisa tu contraseÃ±a > 5'));
        return false;
      }

      dispatch(removeError());
      return true;
  }



  return (
    <>
      <h3 className="auth__title">Register</h3>
      <form onSubmit={handleRegister}>

      {
        msgError &&
        (
        <div className="auth__alert-error">
          {msgError}
        </div>
        )
      }

        <input 
              type="text" 
              placeholder="Name" 
              name="name"
              className="auth__input"
              autoComplete="off"
              value={name}
              onChange={handleInputChange}
          />

          <input
            type="text"
            placeholder="Email"
            name="email"
            className="auth__input"
            autoComplete="off"
            value={email}
            onChange={handleInputChange}
          />
          <input
            type="password"
            placeholder="Password"
            name="password"
            className="auth__input"
            autoComplete="off"
            value={password}
            onChange={handleInputChange}
          />
          <input
            type="password"
            placeholder="Confirm your password"
            name="password2"
            className="auth__input"
            autoComplete="off"
            value={password2}
            onChange={handleInputChange}
          />

          <button type="submit" className="btn btn-primary btn-block mb-5">
            Register
          </button>

          <Link to="/auth/login" className="link">
            Already registered?
          </Link>
      </form>
    </>
  );
};
