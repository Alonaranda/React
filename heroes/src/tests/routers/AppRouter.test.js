import {mount} from 'enzyme';
import React from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { AppRouter } from '../../routers/AppRouter';

describe('Probando AppRouter', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    };

    test('Mostrar loggin si no estoy authenticado', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter/>
            </AuthContext.Provider>
        );
        //console.log(wrapper.html());

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar el componente de marvel, auth = true ', () => {
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Christian'
            }
        };

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter/>
            </AuthContext.Provider>
        );
        console.log(wrapper.html());
        expect(wrapper.find('.navbar').exists()).toBe(true);
        
    });
});