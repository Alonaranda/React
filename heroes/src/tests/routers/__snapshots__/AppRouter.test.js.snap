// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando AppRouter Mostrar loggin si no estoy authenticado 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
